app/services/DFS.coffee

```diff
@@ -0,0 +1,78 @@
+Tuple    = require "../models/Tuple"
+TupleSet = require "../models/TupleSet"
+
+class DFS
+
+  @isValidSeed: (x, y) ->
+    if x - 1 < 0 or x + 1 >= @grid.dimension
+      return false
+    colCount = []
+    for col in [x - 1, x, x + 1]
+      count = 0
```

Comment:
this method took a while to understadnd.  Maybe call `count` something more descriptive, like `numEmpty` and then `colCount` could be `numEmptyInColumn`